File: MinMaxInArray.java
// Time Complexity: O(n)
// Space Complexity: O(n)

public class MinimumAndMaximum{
    public static void Main(String args[]){
        int arr[] = {12,30,10,2,89,10};
        int min= arr[0];
        int max= arrr[0];
        
        for(int i=0;i<arr.length;i++){
            if(arr[i]<min){
                min = arr[i];
            }
             if(arr[i]>max){
                max = arr[i];
            }
        }
        
        System.out.println("Minimum number in array: " + min);
        System.out.println("Maximum number in array: " + max);
    }
}

File: minMaxInArray.cs
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode :
// Any problem you faced while coding this :
using System;

namespace maxMinInArray
{
   class HelloWorld {
    static void Main() {
        int[] a = new int[]{-1, 2, 6, -5, 3, 1};
        int max, min;
        int index = 0;
        
        if(a.Length % 2 == 0) {
         
           if(a[0] > a[1]) {
               max = a[0];
               min = a[1];
           } else {
               max = a[1];
               min = a[0];
           }
           index = 2;
            
        } else {
            max = a[0];
            min = a[0];
            index = 1;
        }
        
        for(int i=index; i < a.Length; i = i + 2) {
            
            if(a[i] > a[i+1]) {
                max = Math.Max(a[i], max);
                min = Math.Min(a[i+1], min);
            } else {
                 min = Math.Min(a[i], min);
                 max = Math.Max(a[i+1], max);
            }
        }
        
        System.Console.WriteLine(min);
          System.Console.WriteLine(max);
    }
}
}
